### PR TITLE
Expose is_abbrev in the mappings apis

### PR DESCRIPTION
**Pull request by leiserfg**

### CODE REVIEW

File: src/nvim/api/keysets.lua [hunk 1]
```diff
@@ -50,6 +50,7 @@ return {
     "callback";
     "desc";
     "replace_keycodes";
+    "is_abbrev";
```
Comment (ii14) [inline src/nvim/api/keysets.lua:53]: "is_" prefix is not necessary
```suggestion
    "abbrev";
```

File: src/nvim/mapping.c [hunk 2]
```diff
@@ -2511,6 +2511,10 @@ void modify_keymap(uint64_t channel_id, Buffer buffer, bool is_unmap, String mod
 #undef KEY_TO_BOOL
   }
   parsed_args.buffer = !global;
+  bool is_abbrev = api_object_to_bool(opts->is_abbrev, "is_abbrev", false, err);
```
Comment (ii14) [inline src/nvim/mapping.c:2514]:
```suggestion
  bool is_abbrev = api_object_to_bool(opts->abbrev, "abbrev", false, err);
```

File: src/nvim/api/keysets.lua (hunk 1 above)
Comment (leiserfg) [reply src/nvim/api/keysets.lua:53]: I just used it to keep compatibility with the internal naming, if that's not a problem, I can commit the suggestion.

Comment (ii14) [reply src/nvim/api/keysets.lua:53]: Nope, how it's internally called is not a problem. Just `abbrev` would correspond to the `:abbrev` command.

**Comment by zeertzjq** [comment]
`abbr` is more consistent with `maparg()` and `mapset()`.

**Comment by leiserfg** [comment]
> `abbr` is more consistent with `maparg()` and `mapset()`.

Well, guys, pick one and I will update the PR with it.

**Comment by leiserfg** [comment]
Do I need to change something else?

**Comment by dundargoc** [comment]
What do you mean lol, all the tests have failed.

**Comment by zeertzjq** [comment]
This can only support `nvim_set_keymap` and `nvim_buf_set_keymap`, but not `nvim_del_keymap` and `nvim_buf_del_keymap` as they do not have `opts` (and this PR makes these two APIs crash immediately). It is a bit strange to have APIs for defining abbreviation but not APIs for removing them.

**Comment by clason** [comment]
The we we need to add `nvim_del_keymap2` that adds an `opts` key and deprecate (and later rename to) `nvim_del_keymap`. Not like we haven't gone down that route before.

**Comment by zeertzjq** [comment]
#23803

**Pull request closed without merging**